File: examples/modulith-clinical-tool-jpa/src/main/java/io/zenwave360/example/clinicaltool/modules/clinical/core/outbound/jpa/PatientRepository.java
package io.zenwave360.example.clinicaltool.modules.clinical.core.outbound.jpa;

import io.zenwave360.example.clinicaltool.modules.clinical.core.domain.*;
import java.math.*;
import java.time.*;
import java.util.*;

import org.jspecify.annotations.NullMarked;
import org.springframework.data.jpa.repository.*;
import org.springframework.stereotype.Repository;

/**
 * Spring Data JPA repository for the Patient entity.
 */
@SuppressWarnings("unused")
@Repository
@NullMarked
public interface PatientRepository extends JpaRepository<Patient, Long> {

    @Override
    @EntityGraph(attributePaths = {"medicalContacts", "personalContacts", "patientAddresses", "hospitalAddresses", "patientWearables"})
    Optional<Patient> findById(Long id);

    @EntityGraph(attributePaths = {"medicalContacts", "personalContacts", "patientAddresses", "hospitalAddresses", "patientWearables"})
    java.util.Optional<Patient> findByPhoneNumberAndHisNumber(String phoneNumber, String hisNumber);
}

File: examples/modulith-klinical-tool-jpa/src/main/kotlin/io/zenwave360/example/clinicaltool/modules/clinical/core/outbound/jpa/PatientRepository.kt
package io.zenwave360.example.clinicaltool.modules.clinical.core.outbound.jpa

import io.zenwave360.example.clinicaltool.modules.clinical.core.domain.Patient
import org.springframework.data.jpa.repository.EntityGraph
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.Optional

/** Spring Data JPA repository for the Patient entity. */
@Suppress("unused")
@Repository
interface PatientRepository : JpaRepository<Patient, Long> {

    @EntityGraph(attributePaths = ["medicalContacts", "personalContacts", "patientAddresses", "hospitalAddresses", "patientWearables"])
    override fun findById(id: Long): Optional<Patient>

    @EntityGraph(attributePaths = ["medicalContacts", "personalContacts", "patientAddresses", "hospitalAddresses", "patientWearables"])
    fun findByIdOrNull(id: Long): Patient? = findById(id).orElse(null)

    @EntityGraph(attributePaths = ["medicalContacts", "personalContacts", "patientAddresses", "hospitalAddresses", "patientWearables"])
    fun findByPhoneNumberAndHisNumber(phoneNumber: String, hisNumber: String): Patient?
}
